js/ui/windowManager.js

```diff
@@ -1401,7 +1401,26 @@ const WindowManager = new Lang.Class({
                 });
                 if (currentSession.length === 0)
                     return false;
+
                 currentSession[0].windowBuilder = window;
+
+                let button = new St.Button({ style_class: 'view-source' });
+                let rect = window.get_frame_rect();
+                button.set_position(rect.x + rect.width - 100, rect.y + rect.height - 100);
+                Main.layoutManager.addChrome(button);
+                let positionChangedId = window.connect('position-changed', Lang.bind(this, function(window) {
+                    let currentSession = this._currentEndlessCodingSessions.filter(function(session) {
+                        return session.windowBuilder === window;
+                    });
+                    if (currentSession.length === 0)
+                        return;
+                    let rect = currentSession[0].windowBuilder.get_frame_rect();
+                    currentSession[0].buttonBuilder.set_position(rect.x + rect.width - 100, rect.y + rect.height - 100);
+                }));
+                let windowsRestackedId = Main.overview.connect('windows-restacked', Lang.bind(this, this._endlessCodingBuilderWindowsRestacked, window));
+                button.connect('clicked', Lang.bind(this, this._endlessCodingBuilderClicked, window));
+                currentSession[0].buttonBuilder = button;
+
```

Comment:
for the future: I wonder whether or not it would be better to have a button in builder itself as opposed to an overlay.

Also, it seems like this code is largely copy-paste from the app-to-builder code earlier. I guess it should probably be refactored so that they share the same code.